pkg/ottl/ottlfuncs/func_replace_all_matches.go

```diff
@@ -43,9 +45,22 @@ func replaceAllMatches[K any](target ottl.PMapGetter[K], pattern string, replace
 		if err != nil {
 			return nil, err
 		}
-		replacementVal, err := replacement.Get(ctx, tCtx)
-		if err != nil {
-			return nil, err
+		if fn.IsEmpty() {
+			replacementVal, err = replacement.Get(ctx, tCtx)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			fnVal := fn.Get()
+			replacementExpr, errNew := fnVal.Get(&FuncArgs[K]{Input: replacement})
```

Comment:
I think we made a mistake when we let `FuncArgs` get added as an exported type.  We should unexport that value and each function should make its own type that can be used as the arguments it expects to be able to hand to the passed in Converter.

Reply:
`FuncArgs` has been declared in this PR in [the file for `replace_pattern`](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27235/files#diff-d32daaa4e1064c4af10a5eb364737ef2bdff7c94d15dbd94a796ddb42ec2afa8R21), so thankfully I think we're okay here.

@rnishtala-sumo We should rename `FuncArgs` to `replaceFuncArgs` or similar to unexport it and make sure it doesn't conflict with any other args structs in the `ottlfuncs` package. I think it should be fine to share it between the four functions, but it's also fine if each function declares its own struct.

Reply:
@evan-bradley we do also export one from here tho: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b5e9e5e4c9c41929b7a70bbe83377a4d30ec43d4/pkg/ottl/expression_test.go#L705

Reply:
@rnishtala-sumo we should do what @evan-bradley mentions tho.

Reply:
yes I'll rename it and ensure that there are no conflicts.

Reply:
I've used names like `replacePatternFuncArgs`, `replaceAllPatternFuncArgs` in each function